File: frontend/src/types/homeDashboard.ts
export interface HomeMetric {
  key: string;
  title: string;
  value: string;
  note: string;
  status: "normal" | "warning" | "danger" | string;
}

export interface HomeMetricSection {
  key: string;
  title: string;
  metrics: HomeMetric[];
}

export interface HomeAiTip {
  category: string;
  title: string;
  content: string;
  level: "normal" | "medium" | "high" | string;
}

export interface HomeDashboard {
  period: string;
  company_name: string;
  sections: HomeMetricSection[];
  ai_tips: HomeAiTip[];
}

File: backend/app/models/home_dashboard.py
from pydantic import BaseModel


class HomeMetric(BaseModel):
    key: str
    title: str
    value: str
    note: str
    status: str = "normal"


class HomeMetricSection(BaseModel):
    key: str
    title: str
    metrics: list[HomeMetric]


class HomeAiTip(BaseModel):
    category: str
    title: str
    content: str
    level: str


class HomeDashboardResponse(BaseModel):
    period: str
    company_name: str
    sections: list[HomeMetricSection]
    ai_tips: list[HomeAiTip]
